### PR TITLE
Do some basic validation of Target Features (#7986)

### PR DESCRIPTION
**Pull request by steven-johnson**

### CODE REVIEW

File: src/Target.cpp [hunk 1]
```diff
@@ -788,6 +788,98 @@ void bad_target_string(const std::string &target) {
...
+            SVE,
+            SVE2,
+            VSX,
+            WasmBulkMemory,
+            WasmSatFloatToInt,
+            WasmSignExt,
+            WasmSimd128,
+            WasmThreads,
+        };
+        user_assert(!features_any_of(bad_features_x86)) << "At least os_entry of the features for "
```
Comment (abadams) [inline src/Target.cpp:816]: "os_entry"?

Comment (steven-johnson) [reply src/Target.cpp]: oops fixed

File: src/Target.cpp [hunk 2]
```diff
@@ -788,6 +788,88 @@ void bad_target_string(const std::string &target) {
...
+            SVE,
+            SVE2,
+            VSX,
+            WasmBulkMemory,
+            WasmSatFloatToInt,
+            WasmSignExt,
+            WasmSimd128,
+            WasmThreads,
+        };
+        user_assert(!features_any_of(bad_features_x86)) << "At least one of the features for "
```
Comment (abadams) [inline src/Target.cpp:811]: A small code change would take the guesswork out of handling this error:
```
for  (Feature f : bad_features_x86) {
   user_assert(!has_feature(f)) << "Target feature " << f ...
}
```

Comment (steven-johnson) [reply src/Target.cpp]: Done

**Comment by abadams** [comment]
Looks like a merge issue with the new wasm feature

**Pull request merged**